cmd/server/main.go

```diff
@@ -198,50 +179,51 @@ func main() {
 	if viper.IsSet("grpc_conn_max_seconds") {
 		grpcKeepAliveFor = time.Second * time.Duration(viper.GetInt("grpc_conn_max_seconds"))
 	}
-	clientBuilder := authorizer.NewClientBuilder(parseClientConfig())
 
 	authorizer, err := authorizer.NewManager(
-		clientBuilder,
+		authorizer.NewClientBuilder(parseClientConfig()),
 		createSystemCache(),
 		createBackendConfig(),
 	)
 	if err != nil {
-		istiolog.Errorf("Unable to create authorizer: %v", err)
-		os.Exit(1)
+		log.Fatalf("Unable to create authorizer: %v", err)
 	}
 
 	adapterConfig := threescale.NewAdapterConfig(authorizer, grpcKeepAliveFor)
 
 	s, err := threescale.NewThreescale(addr, adapterConfig)
-
 	if err != nil {
-		istiolog.Errorf("Unable to start sever: %v", err)
-		os.Exit(1)
-	}
-
-	if version == "" {
-		version = "undefined"
+		log.Fatalf("Unable to start sever: %v", err)
 	}
-	istiolog.Infof("Started server version %s", version)
 
 	shutdown := make(chan error, 1)
 	go func() {
+		if version == "" {
+			version = "undefined"
+		}
+		log.Infof("Starting server version %s", version)
 		s.Run(shutdown)
 	}()
 
 	sigC := make(chan os.Signal, 1)
-	signal.Notify(sigC, syscall.SIGTERM)
+	signal.Notify(sigC, syscall.SIGTERM, syscall.SIGINT)
```

Comment:
So this will also catch Ctrl-C if running interactively. Now that we are here perhaps there are other useful things to do with other signals. Maybe we can also handle SIGHUP for example to force-flush pending data without shutting down.

Reply:
Not sure how useful that will be. Kubernetes sends the pod a sigterm in order to allow it to handle shutting down gracefully. I added the SIGINT there to allow for testing the behaviour but I think that could be removed just as easily as it will serve no purpose in the environment it will live in, wdyt?

Reply:
I think it is still useful to have them, even if it's just for development purposes.